debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -208,6 +209,29 @@ public void execute(ChangeEventSourceContext context, OraclePartition partition,
                             endScn = deviatedScn.get();
                         }
 
+                        final Scn minOpenRedoThreadLastScn = jdbcConnection.getRedoThreadState()
+                                .getThreads()
+                                .stream()
+                                .filter(RedoThread::isOpen)
+                                .map(RedoThread::getLastRedoScn)
+                                .min(Scn::compareTo)
+                                .orElse(Scn.NULL);
+
+                        if (!minOpenRedoThreadLastScn.isNull()) {
+                            if (minOpenRedoThreadLastScn.compareTo(endScn) < 0) {
+                                // There are situations where on first start-up that the startScn may be higher
+                                // than the last flushed redo thread SCN, in which case we should delay by one
+                                // iteration until the startScn is before the minOpenRedoTheadLastScn
+                                if (minOpenRedoThreadLastScn.compareTo(startScn) < 0) {
+                                    pauseBetweenMiningSessions();
+                                    continue;
+                                }
+                                LOGGER.info("Minimum Redo Log Flush SCN is {}, using range {} to {} instead of {} to {}.", minOpenRedoThreadLastScn, startScn,
```

Comment:
This needs to be changed to `debug`, will do so after testing this change with the tests.  I think it also makes sense to refactor this logminer range logic into a separate class, but I'll do that in a separate commit as well.

Reply:
+1 for refactoring!